packages/guides/src/ReferenceResolvers/AnchorReferenceResolver.php

```diff
@@ -46,7 +46,7 @@ public function resolve(LinkInlineNode $node, RenderContext $renderContext, Mess
             return false;
         }
 
-        $node->setUrl($this->urlGenerator->generateCanonicalOutputUrl($renderContext, $target->getDocumentPath(), $target->getAnchor()));
+        $node->setUrl($this->urlGenerator->generateCanonicalOutputUrl($renderContext, $target->getDocumentPath(), $target->getPrefix() . $target->getAnchor()));
```

Comment:
This seems a bug to me, as you already prefix the anchor in the `getAnchor`?

Reply:
The internal target saves the prefix separately. Within rst we do not use the prefixes for references, only in the rendering of links and ids they are prefixed